content/design-systems-for-developers/react/zh-CN/document.md

```diff
@@ -0,0 +1,412 @@
+---
+title: '写给干系人的文档'
+tocTitle: '文档'
+description: '通过文档推动设计系统的使用'
+commit: a032b50
+---
+
+[专业的](https://product.hubspot.com/blog/how-to-gain-widespread-adoption-of-your-design-system) [前端](https://segment.com/blog/driving-adoption-of-a-design-system/) [团队](https://medium.com/@didoo/measuring-the-impact-of-a-design-system-7f925af090f7) 通过采用率来衡量设计系统的成功程度。为了简化所有人的工作量，组件必须是灵活通用的，否则有什么意义呢？
+
+在本章中，我们将创建一个设计系统的“用户手册”来帮助干系人在他们的应用程序中复用组件。在此过程中，我们将揭示 Shopify、Microsoft、Auth0 和 英国政府使用 UI 文档的最佳实践。
+
+![Generate docs with Storybook automatically](/design-systems-for-developers/design-system-generate-docs.jpg)
+
+## 编写详尽的文档
+
+很明显，文档对于 UI 协作开发非常关键，它帮助团队理解如何去使用常用的 UI 组件，但是为什么它需要很多时间去做呢？
+
+如果您从没有创建过文档，您可能会花很多时间忙于非文档内容的部分，比如网站基础架构或和关于技术名词如何写的争论。而且，即使您有时间发布这些文档，在持续开发功能的同时去维护它也很麻烦。
+
+**大多数文档在创建时就已经过时了**，过时的文档破坏了开发人员对设计系统的信任，所以开发人员选择创建新组建而不是重用现有的。
+
+## 需求
+
+我们的文档必须通过以下几点来克服创建和维护文档的固有痛点：
+
+- **🔄 时刻保持最新版本** 通过使用最新生产环境上的代码
+- **✍️ 易于编写** 使用例如：Markdown 这样熟悉的写作工具
+- **⚡️ 减少维护时间** 这样团队可以更专于写作
+- **📐 提供样板** 这样开发人员不需要重写一些通用的模板
+- **🎨 提供可定制** 来适用于极其复杂的用例和组件
+
+作为 Storybook 的用户，我们具有领先优势，因为组件的各个变体已经被记录为 stories（一种文档形式）。一个 story 展示了组件在不同输入下应该如何工作， story 利于编写，且因其引用了生产环境的组件，所以可以自动更新。并且，我们可以使用上一章中的工具对 story 进行回归测试！
+
+> 当您编写 story 时您将免费获得组件的属性文档和使用用例 – Justin Bennett, Engineer at Artsy
+
+## 用 story 生成文档
+
+借助于 Storybook 的文档插件，我们可以从现有的 stories 中生成丰富的文档，它可以减少您的维护时间且默认情况下开箱即用。首先，跳转到您的设计系统根目录下，我们将安装文档插件：
+
+```bash
+yarn add --dev @storybook/addon-docs
+```
+
+在文件 `.storybook/main.js` 中添加插件:
+
+```javascript
+// .storybook/main.js
+
+module.exports = {
+  stories: ['../src/**/*.stories.js'],
+  addons: [
+    '@storybook/preset-create-react-app',
+    '@storybook/addon-actions',
+    '@storybook/addon-links',
+    '@storybook/addon-storysource',
+    '@storybook/addon-knobs',
+    {
+      name: '@storybook/addon-docs',
+      options: {
+        configureJSX: true,
+      },
+    },
+  ],
+};
+```
+
+您应该可以看到您的 Storybook 中有两个选项卡，"Canvas" 选项卡是您的组件开发环境，"Docs" 选项卡是您组件的文档。
+
+![Storybook docs tab](/design-systems-for-developers/storybook-docs.png)
+
+Storybook Docs 插件在运行时帮您的每个组件创建了一个 "Docs" 选项卡。它在您的选项卡中自动填充了常用的文档，例如预览交互式组件，源代码查看和组件属性列表。您可以在 Shopify 和 Auth0 的设计系统正找到类似的功能，所有的设置在 5 分钟内即可完成。
+
+## 扩展文档
+
+目前为止我们只花费了一点点力气就取得了这么大的进步，但是，文档仍然缺少一点“人情味”。我们需要为开发人员提供更多的背景信息（原因、时间和方式）。
+
+首先添加更多的元数据来阐述组件到底是做什么的。 在文件 `src/Avatar.stories.js` 中添加一个副标题来阐述 Avatar 组件是做什么的：
+
+```javascript
+// src/Avatar.stories.js
+
+export default {
+  title: 'Design System|Avatar',
+
+  parameters: {
+    component: Avatar,
+    componentSubtitle: 'Displays an image that represents a user or organization',
+  },
+};
+```
+
+然后给 Avatar 添加 JSDoc（在文件 `src/components/Avatar.js` 中），它为该组件提供了相应的描述。
+
+```javascript
+// src/components/Avatar.js
```

Comment:
Hey! I just noticed this reference to a `components` directory and created [a PR to remove it](https://github.com/chromaui/learnstorybook.com/pull/508). I would remove it here and a handful of times where it appears in code further down in this document.